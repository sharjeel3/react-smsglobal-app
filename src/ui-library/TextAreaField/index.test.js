import React from 'react';
import { mount } from 'enzyme';
import { TextAreaField } from './index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('TEXT AREA FIELD - UI Library', () => {
  it('renders correctly', () => {
    const onChangeMock = jest.fn();
    const tree = renderer
      .create(<TextAreaField id="test" onChange={onChangeMock} value="val" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct value', () => {
    const value = 'some-val';
    const textInput = mount(<TextAreaField id="testing" value={value} />);
    expect(textInput.find('textarea').prop('value')).toBe(value);
  });

  it('should call change handler', () => {
    const mockFn = jest.fn();
    const value = 'new-val';
    const textInput = mount(<TextAreaField onChange={mockFn} id="test" value="" />);
    textInput.find('textarea').simulate('change', { target: { value } });
    expect(mockFn).toBeCalledWith(value);
  });
});
