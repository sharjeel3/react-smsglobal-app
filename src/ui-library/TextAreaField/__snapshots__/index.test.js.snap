// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEXT AREA FIELD - UI Library renders correctly 1`] = `
.c1 {
  font-weight: 600;
  font-size: 1em;
  line-height: 2;
  color: #1C1C1C;
  display: block;
}

.c3 {
  font-size: 1em;
  line-height: 2em;
  color: #EB5757;
  min-height: 2em;
}

.c0 {
  margin-bottom: 0.5em;
}

.c2 {
  font-size: 1em;
  padding: 1em;
  background-color: #eeeeee;
  border: 0.1em solid transparent;
  border-radius: 0.25em;
  color: #1C1C1C;
  height: 8em;
  line-height: 1.5em;
  width: 100%;
  max-width: 100%;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="test"
  >
    
  </label>
  <textarea
    className="c2"
    id="test"
    onChange={[Function]}
    type="text"
    value="val"
  />
  <div
    className="c3"
  >
    
  </div>
</div>
`;
